linera-core/src/node.rs

```diff
@@ -103,12 +103,15 @@ pub trait ValidatorNode {
     async fn upload_blob(&self, content: BlobContent) -> Result<BlobId, NodeError>;
 
     /// Uploads the blobs to the validator.
-    fn upload_blobs(&self, blobs: Vec<Blob>) -> impl futures::Future<Output = Result<(), NodeError>>
-    where
-        Self: Sync,
-    {
-        async {
-            let tasks = blobs.into_iter().map(|blob| self.upload_blob(blob.into()));
+    fn upload_blobs(
```

Comment:
Does someone know why we're not using `async` here? (Without a comment, it's a bit hard to see)

Reply:
It doesn't compile with `async` - something about default implementations in a trait doesn't work with async functions. I'll add a comment.